docs/data/base/components/slider/RangeSlider.tsx

```diff
@@ -60,98 +51,103 @@ const grey = {
   }
 
-  & .${sliderClasses.rail} {
-    display: block;
-    position: absolute;
-    width: 100%;
-    height: 4px;
-    border-radius: 6px;
-    background-color: currentColor;
-    opacity: 0.3;
+  &[data-disabled] {
+    cursor: not-allowed;
   }
 
-  & .${sliderClasses.track} {
-    display: block;
-    position: absolute;
-    height: 4px;
-    border-radius: 6px;
-    background-color: currentColor;
+  .dark &::before {
+    background-color: ${grey[700]};
+  }
+`;
+
+const SliderThumb = styled(BaseSlider.Thumb)`
+  position: absolute;
+  width: 16px;
+  height: 16px;
+  box-sizing: border-box;
+  border-radius: 50%;
+  background-color: black;
+  transform: translateX(-50%);
+  touch-action: none;
+
+  &:focus-visible {
+    outline: 2px solid black;
+    outline-offset: 2px;
   }
 
-  & .${sliderClasses.thumb} {
-    display: flex;
-    align-items: center;
-    justify-content: center;
-    position: absolute;
-    margin-left: -6px;
-    width: 20px;
-    height: 20px;
-    box-sizing: border-box;
-    border-radius: 50%;
-    outline: 0;
-    background-color: ${theme.palette.mode === 'light' ? blue[500] : blue[400]};
-    transition-property: box-shadow, transform;
-    transition-timing-function: ease;
-    transition-duration: 120ms;
-    transform-origin: center;
-
-    &:hover {
-      box-shadow: 0 0 0 6px ${alpha(
-        theme.palette.mode === 'light' ? blue[200] : blue[300],
-        0.3,
-      )};
-    }
-
-    &.${sliderClasses.focusVisible} {
-      box-shadow: 0 0 0 8px ${alpha(
-        theme.palette.mode === 'light' ? blue[200] : blue[400],
-        0.5,
-      )};
-      outline: none;
-    }
-
-    &.${sliderClasses.active} {
-      box-shadow: 0 0 0 8px ${alpha(
-        theme.palette.mode === 'light' ? blue[200] : blue[400],
-        0.5,
-      )};
-      outline: none;
-      transform: scale(1.2);
-    }
+  .dark & {
+    background-color: ${grey[300]};
   }
 
-  & .${sliderClasses.mark} {
-    position: absolute;
-    width: 10px;
-    height: 10px;
-    border-radius: 99%;
-    background-color: ${theme.palette.mode === 'light' ? blue[200] : blue[900]};
-    top: 44%;
-    transform: translateX(-50%);
+  .dark &:focus-visible {
+    outline-color: ${grey[300]};
+    outline-width: 1px;
+    outline-offset: 3px;
   }
 
-  & .${sliderClasses.markActive} {
-    background-color: ${theme.palette.mode === 'light' ? blue[500] : blue[400]};
+  &[data-dragging='true'] {
+    background-color: pink;
+  }
+
+  &[data-disabled],
+  .dark &[data-disabled] {
+    background-color: ${grey[600]};
+  }
+
+  .dark &[data-dragging='true'] {
+    background-color: pink;
+  }
+`;
+
+// we can't use a <label> element in a range slider since the `for` attribute
+// cannot reference more than one <input> element
+// the html spec doesn't forbid <label> without `for` https://html.spec.whatwg.org/multipage/forms.html#the-label-element
+// but eslint complains by default and a11y validators may complain e.g. WAVE
```

Comment:
> Any slider label should be a <label>. In the range slider examples, the label is a <span>

@colmtuite I left a comment here explaining why - in the demos we can change to use `label` elements but it would require `eslint-disable`-ing the `for` attribute rule, what do you think?